From 143ae73ff0e9054cd534c517e9e4d7d7c67c0d17 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sun, 29 Sep 2024 22:13:13 -0700
Subject: [PATCH 3/3] slurm config rocm sdk builder templates

- following variables needs to be changed
  during install to reflect the real
  userid used during the install and install target directory

  ROCM_SDK_USERID
  ROCM_SDK_INSTALL_PREFIX

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 rocm_etc/cgroup.conf         |   2 +
 rocm_etc/slurm.conf_template | 158 +++++++++++++++++++++++++++++++++++
 rocm_etc/slurmdbd.conf       |  39 +++++++++
 3 files changed, 199 insertions(+)
 create mode 100644 rocm_etc/cgroup.conf
 create mode 100644 rocm_etc/slurm.conf_template
 create mode 100644 rocm_etc/slurmdbd.conf

diff --git a/rocm_etc/cgroup.conf b/rocm_etc/cgroup.conf
new file mode 100644
index 0000000000..07fb6c2af7
--- /dev/null
+++ b/rocm_etc/cgroup.conf
@@ -0,0 +1,2 @@
+CgroupPlugin=disabled
+ConstrainCores=yes
diff --git a/rocm_etc/slurm.conf_template b/rocm_etc/slurm.conf_template
new file mode 100644
index 0000000000..1271bf5bfc
--- /dev/null
+++ b/rocm_etc/slurm.conf_template
@@ -0,0 +1,158 @@
+# slurm.conf file generated by configurator.html.
+# Put this file on all nodes of your cluster.
+# See the slurm.conf man page for more information.
+#
+SlurmctldHost=localhost
+AuthType=auth/none
+CredType=cred/none
+#SlurmctldHost=
+#
+#DisableRootJobs=NO
+#EnforcePartLimits=NO
+#Epilog=
+#EpilogSlurmctld=
+#FirstJobId=1
+#MaxJobId=999999
+#GresTypes=
+#GroupUpdateForce=0
+#GroupUpdateTime=600
+#JobFileAppend=0
+#JobRequeue=1
+#JobSubmitPlugins=1
+#KillOnBadExit=0
+#LaunchType=launch/slurm
+#Licenses=foo*4,bar
+#MailProg=/bin/mail
+#MaxJobCount=5000
+#MaxStepCount=40000
+#MaxTasksPerNode=128
+MpiDefault=pmix
+#MpiParams=ports=#-#
+#PluginDir=
+#PlugStackConfig=
+#PrivateData=jobs
+ProctrackType=proctrack/linuxproc
+#Prolog=
+#PrologFlags=
+#PrologSlurmctld=
+#PropagatePrioProcess=0
+#PropagateResourceLimits=
+#PropagateResourceLimitsExcept=
+#RebootProgram=
+ReturnToService=1
+#SallocDefaultCommand=
+SlurmctldPidFile=ROCM_SDK_INSTALL_PREFIX/var/run/slurm/slurmctld.pid
+SlurmctldPort=6817
+SlurmdPidFile=ROCM_SDK_INSTALL_PREFIX/var/run/slurm/slurmd_%n.pid
+SlurmdPort=6818
+SlurmdSpoolDir=ROCM_SDK_INSTALL_PREFIX/var/spool/slurmd_%n
+SlurmUser=ROCM_SDK_USERID
+SlurmdUser=ROCM_SDK_USERID
+#SrunEpilog=
+#SrunProlog=
+StateSaveLocation=ROCM_SDK_INSTALL_PREFIX/var/spool/slurmctld
+SwitchType=switch/none
+#TaskEpilog=
+#TaskPlugin=task/affinity,task/cgroup
+TaskPlugin=task/affinity
+TaskPluginParam=Verbose
+#TaskProlog=
+#TopologyPlugin=topology/tree
+#TmpFS=/tmp
+#TrackWCKey=no
+#TreeWidth=
+#UnkillableStepProgram=
+#UsePAM=0
+#
+#
+# TIMERS
+#BatchStartTimeout=10
+#CompleteWait=0
+#EpilogMsgTime=2000
+#GetEnvTimeout=2
+#HealthCheckInterval=0
+#HealthCheckProgram=
+InactiveLimit=0
+KillWait=30
+#MessageTimeout=10
+#ResvOverRun=0
+MinJobAge=300
+#OverTimeLimit=0
+SlurmctldTimeout=120
+SlurmdTimeout=300
+#UnkillableStepTimeout=60
+#VSizeFactor=0
+Waittime=0
+#
+#
+# SCHEDULING
+#DefMemPerCPU=0
+#MaxMemPerCPU=0
+#SchedulerTimeSlice=30
+SchedulerType=sched/backfill
+SelectType=select/cons_tres
+SelectTypeParameters=CR_Core
+#
+#
+# JOB PRIORITY
+#PriorityFlags=
+#PriorityType=priority/basic
+#PriorityDecayHalfLife=
+#PriorityCalcPeriod=
+#PriorityFavorSmall=
+#PriorityMaxAge=
+#PriorityUsageResetPeriod=
+#PriorityWeightAge=
+#PriorityWeightFairshare=
+#PriorityWeightJobSize=
+#PriorityWeightPartition=
+#PriorityWeightQOS=
+#
+#
+# LOGGING AND ACCOUNTING
+#AccountingStorageEnforce=0
+#AccountingStorageHost=
+#AccountingStorageLoc=
+#AccountingStoragePass=
+#AccountingStoragePort=
+AccountingStorageType=accounting_storage/none
+#AccountingStorageUser=
+#AccountingStoreFlag=
+ClusterName=cluster
+#DebugFlags=
+#JobCompHost=
+#JobCompLoc=
+#JobCompPass=
+#JobCompPort=
+JobCompType=jobcomp/none
+#JobCompUser=
+#JobContainerType=job_container/none
+JobAcctGatherFrequency=30
+JobAcctGatherType=jobacct_gather/none
+SlurmctldDebug=info
+#SlurmctldLogFile=
+SlurmdDebug=info
+SlurmdLogFile=ROCM_SDK_INSTALL_PREFIX/var/log/slurm/slurmd_%n.log
+#SlurmSchedLogFile=
+#SlurmSchedLogLevel=
+#
+#
+# POWER SAVE SUPPORT FOR IDLE NODES (optional)
+#SuspendProgram=
+#ResumeProgram=
+#SuspendTimeout=
+#ResumeTimeout=
+#ResumeRate=
+#SuspendExcNodes=
+#SuspendExcParts=
+#SuspendRate=
+#SuspendTime=
+#
+#
+# COMPUTE NODES
+#NodeName=rocmsdk1 NodeHostname=localhost Port=6818 CPUs=3
+#PartitionName=debug Nodes=rocmsdk1 Default=YES MaxTime=INFINITE State=UP
+
+NodeName=rocmsdk1 NodeHostname=localhostslurm1 Port=6818 CPUs=3 State=UNKNOWN
+NodeName=rocmsdk2 NodeHostname=localhostslurm2 Port=6819 CPUs=3 State=UNKNOWN
+PartitionName=debug Nodes=rocmsdk1,rocmsdk2 Default=YES MaxTime=INFINITE State=UP
diff --git a/rocm_etc/slurmdbd.conf b/rocm_etc/slurmdbd.conf
new file mode 100644
index 0000000000..0e1db76edf
--- /dev/null
+++ b/rocm_etc/slurmdbd.conf
@@ -0,0 +1,39 @@
+#
+# Example slurmdbd.conf file.
+#
+# See the slurmdbd.conf man page for more information.
+#
+# Archive info
+#ArchiveJobs=yes
+#ArchiveDir="/tmp"
+#ArchiveSteps=yes
+#ArchiveScript=
+#JobPurge=12
+#StepPurge=1
+#
+# Authentication info
+AuthType=auth/none
+#AuthInfo=/var/run/munge/munge.socket.2
+#
+# slurmDBD info
+DbdAddr=localhost
+DbdHost=localhost
+#DbdPort=7031
+SlurmUser=slurm
+#MessageTimeout=300
+DebugLevel=verbose
+#DefaultQOS=normal,standby
+LogFile=/opt/rocm_sdk_612/var/log/slurm/slurmdbd.log
+PidFile=/opt/rocm_sdk_612/run/slurm/slurmdbd.pid
+#PluginDir=/usr/lib/slurm
+#PrivateData=accounts,users,usage,jobs
+#TrackWCKey=yes
+#
+# Database info
+StorageType=accounting_storage/mysql
+#StorageHost=localhost
+#StoragePort=1234
+StoragePass=password
+StorageUser=slurm
+#StorageLoc=slurm_acct_db
+
-- 
2.41.1

